Frameworks/Foundation/NSProxy.mm

```diff
@@ -14,220 +14,317 @@
 //
 //******************************************************************************
 
-#import <StubReturn.h>
 #import <Foundation/NSProxy.h>
+#import <Starboard/SmartTypes.h>
+
+#import <objc/objc-arc.h>
 
 @implementation NSProxy
+
+#pragma region NSProxy member functions
+
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 + (id)alloc {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self allocWithZone:nil];
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 + (id)allocWithZone:(NSZone*)zone {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return class_createInstance(self, 0);
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (void)dealloc {
-    UNIMPLEMENTED();
+    object_dispose(self);
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (void)finalize {
-    UNIMPLEMENTED();
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (void)forwardInvocation:(NSInvocation*)anInvocation {
-    UNIMPLEMENTED();
+    [NSException raise:NSInvalidArgumentException
+                format:@"[NSProxy forwardInvocation:] was called directly. It must be overridden in a subclass."];
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (NSMethodSignature*)methodSignatureForSelector:(SEL)aSelector {
-    UNIMPLEMENTED();
-    return StubReturn();
+    [NSException raise:NSInvalidArgumentException
+                format:@"[NSProxy methodSignatureForSelector:] was called directly. It must be overridden in a subclass."];
+    return nil;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 + (BOOL)respondsToSelector:(SEL)aSelector {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return class_respondsToSelector(self, aSelector);
+}
+
+// clang format doesn't play well with 'class' here, so turn it off
+// clang-format off
+
+/**
+ @Status Interoperable
+ @Notes
+*/
++ (Class)class {
+    return self;
+}
+
+
+/**
+ @Status Interoperable
+*/
++ (Class)superclass {
+    return class_getSuperclass(self);
+}
+
+// clang-format on
+
+/**
+ @Status Interoperable
+*/
+
+- (NSString*)description {
+    return [NSString stringWithFormat:@"<%@: %p>", NSStringFromClass([self class]), self];
+}
+
+/**
+ @Status Interoperable
+*/
+- (NSString*)debugDescription {
+    return [self description];
 }
 
+#pragma endregion
+
+#pragma region NSObject protocol functions
+
+// clang format doesn't play well with 'class' here, so turn it off
+// clang-format off
+
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
-+ (Class) class {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (Class)class {
+    return object_getClass(self);
 }
 
-    /**
-     @Status Stub
-     @Notes
-    */
-    - (Class) class {
-    UNIMPLEMENTED();
-    return StubReturn();
+/**
+ @Status Interoperable
+ @Notes
+*/
+- (Class)superclass {
```

Comment:
Mm, my thought is that if a change like that breaks something, we'll either be able to catch it in CI build, or it's so incredibly subtle despite being in such an exercised class that doing it in a separate change won't really gain us anything, so I'll elect to risk it. (I will probably regret saying this a week from now)